velox/expression/CoalesceExpr.cpp

```diff
@@ -107,14 +108,48 @@ TypePtr CoalesceCallToSpecialForm::resolveType(
   return CoalesceExpr::resolveType(argTypes);
 }
 
+ExprPtr CoalesceCallToSpecialForm::optimize(
+    std::vector<ExprPtr>& compiledChildren) {
+  auto numChildren = compiledChildren.size();
+  std::vector<column_index_t> nullIndices;
+  for (auto i = 0; i < numChildren; i++) {
```

Comment:
Do we also optimize coalesce(a, null, b, 5, c, d) -> coalesce(a, b, 5)? Let's add a test.